sdk/identity/Azure.Identity/src/ImdsManagedIdentityProbeSource.cs

```diff
@@ -69,13 +67,16 @@ protected override Request CreateRequest(string[] scopes)
 
             request.Uri.AppendQuery("resource", resource);
 
-            if (!string.IsNullOrEmpty(_clientId))
+            string idQueryParam = _managedIdentityId?._idType switch
             {
-                request.Uri.AppendQuery(Constants.ManagedIdentityClientId, _clientId);
-            }
-            if (!string.IsNullOrEmpty(_resourceId))
+                ManagedIdentityIdType.ClientId => Constants.ManagedIdentityClientId,
+                ManagedIdentityIdType.ResourceId => Constants.ManagedIdentityResourceId,
```

Comment:
Should this also handle `ObjectId`?

Reply:
I'm not sure it matters for the probe request to parse these at all, now that I think about it. But it's probably not a bad idea to add just in case.